File: golden-paths/ruby/rails/test/contract_test.rb
# frozen_string_literal: true

require_relative "test_helper"

# Golden-path contract tests — drive the REAL app through the full Rails stack (router + controllers).
class ContractTest < ActionDispatch::IntegrationTest
  test "health is ok" do
    get "/health"
    assert_response :success
    assert_match "ok", response.body
  end

  test "ready is ready" do
    get "/ready"
    assert_response :success
    assert_match "ready", response.body
  end

  test "hello returns the known payload" do
    get "/hello"
    assert_response :success
    assert_match "hello, weyland", response.body
    assert_match "golden-ruby-rails", response.body
  end

  test "metrics exposes prometheus" do
    get "/metrics"
    assert_response :success
    assert_match "golden_hello_requests", response.body
  end
end

File: golden-paths/cpp/httplib/test/contract_test.cpp
#include <doctest/doctest.h>

#include <string>

#include "handlers.hpp"

TEST_CASE("health is ok") {
  CHECK(golden::health_json().find("ok") != std::string::npos);
}

TEST_CASE("ready is ready") {
  CHECK(golden::ready_json().find("ready") != std::string::npos);
}

TEST_CASE("hello returns the known payload") {
  const auto body = golden::hello_json();
  CHECK(body.find("hello, weyland") != std::string::npos);
  CHECK(body.find("golden-cpp-httplib") != std::string::npos);
}

TEST_CASE("metrics exposes prometheus") {
  CHECK(golden::metrics_text().find("golden_hello_requests") != std::string::npos);
}
